internal/fileutil/fileutil.go

```diff
@@ -46,6 +46,11 @@ func UpdateFileIfDifferent(path string, content []byte) (bool, error) {
 	if err := f.Close(); err != nil {
 		return false, err
 	}
+	// preserve permissions
+	// silently ignore failure to avoid breaking changes
+	if fileInfo, err := os.Stat(path); err == nil {
+		os.Chmod(f.Name(), fileInfo.Mode())
```

Comment:
It's nice to have an explicit discard when intentionally ignoring errors:

```suggestion
		_ = os.Chmod(f.Name(), fileInfo.Mode())
```